server/src/users.js

```diff
@@ -102,9 +102,9 @@ exports.registerLogin = function(deviceId, data, canUpdate) {
   }
   const secretHashed = createHash(data.secret);
   return db.insert(
-    `INSERT INTO devices (id, secret_hashed)
-     VALUES ($1, $2)`,
-    [deviceId, secretHashed || null]
+    `INSERT INTO devices (id, secret_hashed, accountid)
+     VALUES ($1, $2, $3)`,
+    [deviceId, secretHashed || null, accountId || null]
   ).then((inserted) => {
     const userAbTests = abTests.updateAbTests({}, getForceAbTests());
     if (inserted) {
```

Comment:
It's definitely not right. I don't think anything uses that codepath, which is why it hasn't come up. At some point the extension held the name/etc editing interface, and it would re-register the user with the new values. Now none of that applies. I've made a follow-up in #5213